libraries/chain/CMakeLists.txt

```diff
@@ -127,11 +127,16 @@ add_library( eosio_chain
              symbol.cpp
              whitelisted_intrinsics.cpp
              thread_utils.cpp
+             peer_keys_db.cpp
              platform_timer_accuracy.cpp
              ${PLATFORM_TIMER_IMPL}
              ${HEADERS}
              )
 
+if(CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 14.0)
+  target_compile_options(eosio_chain PUBLIC -Wthread-safety)
+endif()
```

Comment:
In the future we should consider making this common somewhere and not copy paste it all over.

Also, since this is `PUBLIC` it means everything that links to `libchain` also gets it enabled. This is true today for everything that links to `net_plugin` -- for example nodeos' `main.cpp` currently have `-Wthread-safety` enabled. Is that a problem? If it's not a problem, all the more reason to consider just enabling it globally.

Reply:
Would you like me to add it to the top `CMakeLists.txt`?

Reply:
I wasn't really sure what the side effect was for enabling `-Wthread-safety` on code that isn't prepared for it. We're already doing that a little because of net_plugin and now we're going to be doing it a lot because of libchain. If that's fine maybe we blanket enable it at top anyways, if that's bad then maybe we should change these to `PRIVATE` instead. Doesn't need to be done now.

Reply:
I was just trying it out and it builds fine. Checking it in to see if there is any test issue. I can undo that commit if desired.

Reply:
> I wasn't really sure what the side effect was for enabling `-Wthread-safety` on code that isn't prepared for it.

I believe that it should be fine (a no-op)